serverrpc/logs.go

```diff
@@ -0,0 +1,62 @@
+package serverrpc
+
+import (
+	"bytes"
+	"io"
+	"time"
+
+	timetypes "github.com/docker/engine-api/types/time"
+	"github.com/golang/glog"
+	"github.com/hyperhq/hyperd/daemon"
+	"github.com/hyperhq/hyperd/types"
+)
+
+func (s *ServerRPC) ContainerLogs(req *types.ContainerLogsRequest, stream types.PublicAPI_ContainerLogsServer) error {
+	glog.V(3).Infof("ContainerLogs with request %s", req.String())
+
+	var since time.Time
+	if req.Since != "" {
+		s, n, err := timetypes.ParseTimestamps(req.Since, 0)
+		if err != nil {
+			return err
+		}
+		since = time.Unix(s, n)
+	}
+
+	buffer := bytes.NewBuffer([]byte{})
+
+	var stop <-chan bool
+
+	logsConfig := &daemon.ContainerLogsConfig{
+		Follow:     req.Follow,
+		Timestamps: req.Timestamps,
+		Since:      since,
+		Tail:       req.Tail,
+		UseStdout:  req.Stdout,
+		UseStderr:  req.Stderr,
+		OutStream:  buffer,
+		Stop:       stop,
+	}
+
+	err := s.daemon.GetContainerLogs(req.Container, logsConfig)
+	if err != nil {
+		glog.Errorf("ContainerLogs error: %v", err)
+		return err
+	}
+
+	for {
+		s, err := buffer.ReadString(10)
```

Comment:
No, it will return the whole bytes in buffer,and "err = io.EOF".And I have to process it and return the left bytes without '\n'.But maybe we can just read the logs at once,like `rpc ContainerLogs(ContainerLogsRequest) returns (ContainerLogsResponse) {}`?